### ISSUE TITLE
Compilation error with openssl

### ISSUE THREAD

**Issue by dellarte** [spinkham/skipfish#4]
Hello,

I've obviously installed all dependancies and also set CFLAGS=-I/usr/includes

but i get 2 error while compilation : 

```
cc -L/usr/local/lib/ -L/opt/local/lib src/skipfish.c -o skipfish \
        -O3 -Wno-format -Wall -funsigned-char -g -ggdb -I/usr/local/include/ -I/opt/local/include/ -I/usr/include/ -DVERSION=\"2.10b\" src/http_client.c src/database.c src/crawler.c src/analysis.c src/report.c src/checks.c src/signatures.c src/auth.c src/options.c -lcrypto -lssl -lidn -lz -lpcre
src/http_client.c: In function ‘check_ssl’:
src/http_client.c:1965:10: error: dereferencing pointer to incomplete type ‘SSL_CIPHER’ {aka ‘const struct ssl_cipher_st’}
   if(!(cp->algo_strength & SSL_MEDIUM) && !(cp->algo_strength & SSL_HIGH))
          ^~
src/http_client.c:1982:34: error: dereferencing pointer to incomplete type ‘X509’ {aka ‘struct x509_st’}
     if (ASN1_UTCTIME_cmp_time_t(p->cert_info->validity->notBefore, cur_time)
```

**Comment by dorpvom** [comment]
Hi, had the same problem today.
For me - running ubuntu-mate 18.04 - installing `libssl1.0-dev` via aptitude did the trick. On 18.04 `libssl-dev` defaults to 1.1. Since skipfish works on Kali, which also uses 1.0, I came to trying this.
Maybe this can help you as well.

EDIT: Note that I cloned the kali repo [1]. Don't know if that's newer, older or both are in sync.
[1] git://git.kali.org/packages/skipfish.git
